migration-engine/core/src/commands/diff.rs

```diff
@@ -62,10 +82,53 @@ pub async fn diff(params: DiffParams, host: Arc<dyn ConnectorHost>) -> CoreResul
     Ok(DiffResult { exit_code })
 }
 
+// Grab the preview features and namespaces. Normally, we can only grab these from Schema files,
+// and we usually only expect one of these within a set of DiffTarget.
+// However, in case there's multiple, we union the results. This may be wrong.
+fn namespaces_and_preview_features_from_diff_targets(
```

Comment:
I think we should persist the `TODO` of reading the schema file from disk twice and parsing it twice, it should be a small fun refactoring to fix that.

Reply:
Issue: https://github.com/prisma/schema-team/issues/365